types/types.go

```diff
@@ -128,8 +128,10 @@ type ImageSource interface {
 	GetSignatures(ctx context.Context, instanceDigest *digest.Digest) ([][]byte, error)
 	// LayerInfosForCopy returns either nil (meaning the values in the manifest are fine), or updated values for the layer blobsums that are listed in the image's manifest.
 	// The Digest field is guaranteed to be provided; Size may be -1.
+	// Implementations may ignore desiredLayerCompression and always return
+	// digests for there internal compression format
 	// WARNING: The list may contain duplicates, and they are semantically relevant.
-	LayerInfosForCopy(ctx context.Context) ([]BlobInfo, error)
+	LayerInfosForCopy(ctx context.Context, desiredLayerCompression LayerCompression) ([]BlobInfo, error)
```

Comment:
(`copy.Image` is really the only user of `LayerInfosForCopy`, so if that user needs to call this function twice, the API should just be changed to return two sets of values in a single call.)

Reply:
How about something like the following, I can make the change when I get some positive feedback on the API:

type CopyLayerInfo struct {
  blobInfo BlobInfo
  compressedBlobInfo BlobInfo
}
LayerInfosForCopy(ctx context.Context) ([]CopyLayerInfo, error)

Implementations could choose to populate compressedBlobInfo or not

Reply:
I’d just have `LayerInfosForCopy` return `(blobInfos, uncompressedBlobInfos, error)` so that transports that don’t implement it only need an one-word modification.